app/domain/errors.rb

```diff
@@ -0,0 +1,213 @@
+          code: "CONJ00008E"
+        )
+
+      end
+
+      module RequestBody
+
+        MissingRequestParam = ::Util::ErrorClassWithCode.new(
+          msg: "field '{0}' is missing or empty in request body",
+          code: "CONJ00009E"
+        )
+
+      end
+
+      module AuthnOidc
+
+        ProviderDiscoveryTimeout = ::Util::ErrorClassWithCode.new(
+          msg: "OIDC provider discovery failed with timeout error (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00010E"
+        )
+
+        ProviderDiscoveryFailed = ::Util::ErrorClassWithCode.new(
+          msg: "OIDC provider discovery failed (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00011E"
+        )
+
+        ProviderFetchCertificateFailed = ::Util::ErrorClassWithCode.new(
+          msg: "Failed to fetch certificate from OIDC provider (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00012E"
+        )
+
+        IdTokenFieldNotFoundOrEmpty = ::Util::ErrorClassWithCode.new(
+          msg: "Field '{0}' not found or empty in ID Token",
+          code: "CONJ00013E"
+        )
+
+        IdTokenInvalidFormat = ::Util::ErrorClassWithCode.new(
+          msg: "Invalid ID Token Format (3rdPartyError ='{0}')",
+          code: "CONJ00014E"
+        )
+
+        IdTokenVerifyFailed = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token verification failed (3rdPartyError ='{0}')",
+          code: "CONJ00015E"
+        )
+
+        IdTokenExpired = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token Expired",
+          code: "CONJ00016E"
+        )
+
+        AdminAuthenticationDenied = ::Util::ErrorClassWithCode.new(
+          msg: "admin user is not allowed to authenticate with OIDC",
+          code: "CONJ00017E"
+        )
+
+      end
+
+      module AuthnIam
+
+        InvalidAWSHeaders = ::Util::ErrorClassWithCode.new(
+          msg: "'Invalid or Expired AWS Headers: {0}",
+          code: "CONJ00018E"
+        )
+
+      end
+
+      module AuthnK8s
+
+        WebserviceNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Webservice '{0}' wasn't found",
+          code: "CONJ00019E"
+        )
+
+        HostNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Host '{0}' wasn't found",
+          code: "CONJ00020E"
+        )
+
+        HostNotAuthorized = ::Util::ErrorClassWithCode.new(
+          msg: "'{0}' does not have 'authenticate' privilege on {1}",
+          code: "CONJ00021E"
+        )
+
+        CSRIsMissingSpiffeId = ::Util::ErrorClassWithCode.new(
+          msg: 'CSR must contain SPIFFE ID SAN',
+          code: "CONJ00022E"
+        )
+
+        CSRNamespaceMismatch = ::Util::ErrorClassWithCode.new(
+          msg: "Namespace in SPIFFE ID '{0}' must match namespace implied by common name '{1}'",
+          code: "CONJ00023E"
+        )
+
+        PodNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "No Pod found for podname '{0}' in namespace '{1}'",
```

Comment:
is pod capilatised? Is it a name of a product? I think it should be small letter.
"No pod found for '{0}' in namespace '{1}'" - i also think podname may be superfluous in this message. We can discuss this